### DISCUSSION TITLE
A problem about URL pass parameters

### DISCUSSION THREAD

**Discussion by Jasonzzzzzzzzzzzzzz** [GMOD/jbrowse-components#4551] [Q&A] [answered]
Hi there,
 
When I try to pass parameters by URL, for example "config=dog/config.json&tracklist=true&tracks=canFam6-refseq,canFam6-gff&loc=ZWINT&assembly=Dog10K_Boxer_Tasha", I will a get a error.
![pic1](https://github.com/user-attachments/assets/dbf7fe48-1249-4f26-9f14-5ec372ee057b)
I just pass two tracks name in URL, and you can see, in the picture, the tracklist on the right just show two tracks, but actually there are three tracks in the Jbrowse and the first is error. I notice If I replace loc and use "notation for range", the problem is solved. 
![image](https://github.com/user-attachments/assets/7e03f7e4-6b1a-492c-ae47-4c6af4e9feeb)

However, if i want use genid, how can I solve the problem.

Thanks!

**Comment by cmdcolin** [comment]
hi there, this is a weird issue that you found. unfortunately, I am not able to reproduce this locally yet.

for example, if i have hg19 and hg38 with gene tracks indexed, then using e.g. &loc=PTEN&assembly=hg19 and &loc=PTEN&assembly=hg38 both work fine

even if I create a feature that has a feature named UNIQUENAME that only exists in e.g. hg19 and then try to access it via hg38 e.g. with &loc=UNIQUENAME&assembly=hg38 then this will still doesn't confuse the issue and add the track to the wrong assembly like you are seeing.


if your data directory is not too large, maybe try to zip up the whole thing and send it to me? colin.diesh@gmail.com

what version of JBrowse 2 are you using also?

**Reply by cmdcolin** [reply]
(you can omit large files like the FASTA if you want, i can download or mock those myself for cfam manually)

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
Can I just send a config.json file to you? You can directly visit our data, and if there are some network problems, please tell me, I will zip up the whole thing and send it to you. The version of Jbrowse2 we use is jbrowse-web-v2.11.1 .
[config.json](https://github.com/user-attachments/files/16939137/config.json)

**Reply by cmdcolin** [reply]
the config.json might be a good start, having the gff files and the trix directory would also be good as it would help me reproduce more fully. a link to the live site would be sufficient probably too (in the screenshot it says http://172.16.7.100/Visual/Index ? not sure i can access that)

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
Sure, this is our live stie: 
https://dog10k.kiz.ac.cn/Visual/Index
 e.g. (https://dog10k.kiz.ac.cn/Visual/Index?config=dog/config.json&loc=chr1:6000-7000&assembly=Broad_CanFam_3.1&tracklist=true&tracks=canFam3-refseq,canFam3-gff,canFam3-vcf) this can work fine,if I change a genome version to canfam6 e.g. (https://dog10k.kiz.ac.cn/Visual/Index?config=dog/config.json&tracklist=true&tracks=canFam6-refseq,canFam6-gff&loc=ZXDC&assembly=Dog10K_Boxer_Tasha) it show that error.
If you need anything else, just let me know.
Cheers!

**Reply by cmdcolin** [reply, marked as answer]
I believe what you are seeing is because your config file points to the "canFam3 trix index" but says it belongs to "canFam6/Dog10K_Boxer_Tasha"

specifically the very end of your config
```
{
      "type": "TrixTextSearchAdapter",
      "textSearchAdapterId": "canFam6-index",
      "ixFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Broad_CanFam_3.1.ix",
        "locationType": "UriLocation"
      },
      "ixxFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Broad_CanFam_3.1.ixx",
        "locationType": "UriLocation"
      },
      "metaFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Broad_CanFam_3.1_meta.json"
      },
      "assemblyNames": ["Dog10K_Boxer_Tasha"]<--- this is canfam6, but files seem to indicate it's the trix files are for canfam3.1
    }
```

i guessed the file name and changed it to

```
{
      "type": "TrixTextSearchAdapter",
      "textSearchAdapterId": "canFam6-index",
      "ixFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Dog10K_Boxer_Tasha.ix",
        "locationType": "UriLocation"
      },
      "ixxFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Dog10K_Boxer_Tasha.ixx",
        "locationType": "UriLocation"
      },
      "metaFilePath": {
        "uri": "https://file.kiz.ac.cn/dog10k/Dog10K_Boxer_Tasha_meta.json"
      },
      "assemblyNames": ["Dog10K_Boxer_Tasha"]
    }
```

and it worked :) i was actually able to test this locally because all your files are absolute urls.

hope that helps! the browser looks great!

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
It works! I made a stupid mistake...
Really thanks for your patience and help!
